instance/migrations/0109_alter_json_fields.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.7 on 2018-10-10 20:48
+from __future__ import unicode_literals
+
+from django.db import migrations
+import django.contrib.postgres.fields
+import functools
+import instance.models.utils
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('instance', '0108_manage_own_permission'),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            "ALTER TABLE instance_openedxappserver ALTER COLUMN openstack_server_base_image TYPE JSON USING openstack_server_base_image::JSON;",
```

Comment:
Just wondering here – I expected that Django's built-in `JSONField` uses `JSONB` database fields, whereas this command ssems to set the column type to `JSON`.  See also the [`JSONField` documentation](https://docs.djangoproject.com/en/2.1/ref/contrib/postgres/fields/#jsonfield).

To be clear, I haven't tested whether this works as intended; I'm just wondering why the column type isn't what I expect here.

Reply:
Also in case this fails, do we have/need a migration going backwards?

Reply:
We sometimes have migrations that only work one way, but since it's easy to add the backwards migrations here, I'd be in favour of doing so.

Reply:
@smarnach I'll update it to use JSONB instead and create a backwards migration as well. Thank you!

Reply:
@earaujoassis I'm not sure what is correct, but I believe that only the correct version should work.  So you should be able to find out what right by simply trying it.

Another thing I just noticed – don't we need to modify Django's notion of the models in the migration as well?  Otherwise Django believes we are still using the old type.

I don't really know what the migration code is supposed to look like, so I'm asking you to figure it out. :)

Reply:
@smarnach Django's models using that field were updated as well. We just need to update which class field is being used by the model-mapping, which is done in the following file: `instance/models/openedx_appserver.py`

Reply:
Please correct me if I misunderstood your message. Thank you!

Reply:
@earaujoassis Yeah, my wording was probably confusing.  Django tracks the state of each model in the migrations, and compares the state to what's in the code.  The `RunSQL` directives you added are opaque to Django, so when resolving the state in the migration, it will still end up with the old type of the fields.  So if you run `makemigrations` on your branch, Django still generates `AlterField` directives.

I just did this locally and looked at the SQL output of the automatically generated migrations.  It looks like Django is able to do these migrations automatically, and you don't need the manually written SQL.  I haven't actually tested it, but on the surface it looks promising.

So I believe all you need to do is

* remove the manually written migration.
* run `honcho run ./manage.py makemigrations`.
* test whether it actually works correctly.